File: 800/2121A/main.java
import java.util.*;

public class main 
{
    public static void main(String[] args) 
    {
        Scanner sc = new Scanner(System.in);

        int t = sc.nextInt();

        while (t-- > 0) 
        {
            int n = sc.nextInt();
            int s = sc.nextInt();

            int[] a = new int[n];
            for (int i = 0; i < n; i++)
                a[i] = sc.nextInt();

            if (n == 1) 
                System.out.println(Math.abs(s - a[0]));
            else if (s <= a[0])
                System.out.println(Math.abs(a[n - 1] - s));
            else if (s >= a[n - 1])
                System.out.println(Math.abs(s - a[0]));
            else 
            {
                int left = s - a[0];
                int right = a[n - 1] - s;
                System.out.println(Math.min(left, right) * 2 + Math.max(left, right));
            }
        }
    }
}

File: 800/2121A/main.cpp
#include<bits/stdc++.h>
using namespace std;
 
int main()
{
    int t;
    cin>>t;
    
    while(t--)
    {
        int n,s;
        cin>>n>>s;
        
        int mini = INT_MAX;
        int maxi = INT_MIN;
        
        vector<int> a(n);
        for(int i=0;i<n;i++)
            cin>>a[i];
        
        if(n == 1)
            cout<<abs(s-a[0])<<endl;
        else if(s <= a[0])
            cout<<abs(a[n-1] - s)<<endl;
        else if(s >= a[n-1])
            cout<<abs(s-a[0])<<endl;
        else
            cout<<min(s-a[0], a[n-1]-s)*2 + max(s-a[0], a[n-1]-s)<<endl;
    }
    
    return 0;
}